src/GPX/Writer.php

```diff
@@ -11,6 +12,11 @@ function __construct( Track $track )
 	}
 
 	function writeGpx( string $fileName )
+	{
+		file_put_contents( $fileName, $this->getGpxData() );
+	}
+
+	function getGpxData() : string
```

Comment:
I don't get why you would need this - if you construct a Writer with `$track` already, then you already have the `Track` to call `getTrackData()` on?

Reply:
I guess since you already defined the header (XML) for the GPX file in the `writeGpx` function, I thought about extracting it into its own function.
Writing the result into the filesystem has little benefit to me, so I rather grab the data and send it to s3.
I know I can use the s3 protocol in the `file_put_contents`, but the PR would allow me to do something like this in Laravel:
```
$writer = new Writer($track);

Storage::disk('s3')->put($path, $writer->getGpxData());
```

However, thinking about it, in my opinion moving the `getGpxData` into the Track class and calling it `getGpxDataAsXml` probably makes more sense.

Reply:
Hmm, now I think about it. The responsibility to create XML is what should be in the writer, as at some point I was thinking of allowing it to have multiple segments and/or different GPX types (route, vs track, vs waypoints) as well. So what you did is good - I misunderstood.

Reply:
Awesome! Thank you for consider it!